File: app/src/main/java/wangdaye/com/geometricweather/weather/apis/OpenMeteoAirQualityApi.java
package wangdaye.com.geometricweather.weather.apis;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;
import wangdaye.com.geometricweather.weather.json.openmeteo.OpenMeteoAirQualityResult;

/**
 * Open-Meteo API
 */
public interface OpenMeteoAirQualityApi {

    @GET("v1/air-quality?hourly=pm10,pm2_5")
    Observable<OpenMeteoAirQualityResult> getAirQuality(@Query("latitude") double latitude,
                                                        @Query("longitude") double longitude);
}

File: app/src/main/java/wangdaye/com/geometricweather/weather/apis/OpenMeteoAirQualityApi.kt
package wangdaye.com.geometricweather.weather.apis

import io.reactivex.rxjava3.core.Observable
import retrofit2.http.GET
import retrofit2.http.Query
import wangdaye.com.geometricweather.weather.json.openmeteo.OpenMeteoAirQualityResult

/**
 * Open-Meteo API
 */
interface OpenMeteoAirQualityApi {
    @GET("v1/air-quality?hourly=pm10,pm2_5")
    fun getAirQuality(
        @Query("latitude") latitude: Double,
        @Query("longitude") longitude: Double
    ): Observable<OpenMeteoAirQualityResult>
}